File: 2016/fase1/lâmpadas-do-hotel/lâmpadas-do-hotel.java
import java.util.Scanner;

public class LampadasDoHotel {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Lendo os estados iniciais e finais das lâmpadas
        int IA = scanner.nextInt();
        int IB = scanner.nextInt();
        int FA = scanner.nextInt();
        int FB = scanner.nextInt();
        
        // Contando o número mínimo de vezes que os interruptores devem ser apertados
        int switchesPressed = countSwitchesPressed(IA, IB, FA, FB);
        
        // Imprimindo o resultado
        System.out.println(switchesPressed);
        
        scanner.close();
    }
    
    public static int countSwitchesPressed(int IA, int IB, int FA, int FB) {
        // Verificando se as lâmpadas já estão nos estados desejados
        if (IA == FA && IB == FB) {
            return 0;
        }
        
        // Verificando se é necessário apenas apertar um dos interruptores
        if (IA != FA && IB != FB) {
            return 2;
        }
        
        // Se apenas uma lâmpada precisa mudar de estado, retornamos 1
        return 1;
    }
}

File: 2016/fase1/lâmpadas-do-hotel/lâmpadas-do-hotel.py
def min_interruptores(IA, IB, FA, FB):
    # Verificar se os estados iniciais e finais das lâmpadas são os mesmos
    if IA == FA and IB == FB:
        return 0
    
    # Verificar se é necessário inverter o estado de uma das lâmpadas com C1
    if IA != FA and IB != FB:
        return 1
    
    # Se apenas uma lâmpada precisa ser invertida com C1
    if IA != FA or IB != FB:
        return 1
    
    # Se apenas uma lâmpada precisa ser invertida com C2
    return 2

# Lendo os estados iniciais e finais das lâmpadas
IA, IB, FA, FB = map(int, input().split())

# Chamando a função e imprimindo o resultado
print(min_interruptores(IA, IB, FA, FB))